plugins/ExamplePlugin/src/SumOfExponential.py

```diff
@@ -0,0 +1,114 @@
+        if "varName" not in child.attrib:
+          raise IOError("ExamplePlugin: attribute varName must be present in <coefficient> XML node!")
+        container.coefficients[child.attrib['varName']] = float(child.text)
+      if child.tag.strip() == "startMonotonicVariableValue":
+        container.startValue = float(child.text)
+      if child.tag.strip() == "endMonotonicVariableValue":
+        container.endValue = float(child.text)
+      if child.tag.strip() == "numberCalculationPoints":
+        container.numberPoints = int(child.text)
+    if container.startValue is None:
+      raise IOError("ExamplePlugin: <startMonotonicVariableValue> XML has not been inputted!")
+    if container.endValue is None:
+      raise IOError("ExamplePlugin: <endMonotonicVariableValue> XML has not been inputted!")
+    container.variables.pop(container.variables.index("Xi"))
+    container.variables.pop(container.variables.index("monotonicVariable"))
+
+  def initialize(self, container,runInfoDict,inputFiles):
+    """
+      Method to initialize this plugin
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, runInfoDict, dict, dictionary containing all the RunInfo parameters (XML node <RunInfo>)
+      @ In, inputFiles, list, list of input files (if any)
+      @ Out, None
+    """
+    for var in container.variables:
+      if var not in container.coefficients:
+        container.coefficients[var] = 1.0
+        print("ExamplePlugin: not found coefficient for variable "+var+". Default value is 1.0!")
+    container.stepSize = (container.endValue - container.startValue)/float(container.numberPoints)
+
+  def run(self, container, Inputs):
+    """
+      This is a simple example of the run method in a plugin.
+      This method takes the variables in input and computes
+      oneOutputOfThisPlugin(t) = var1Coefficient*exp(var1*t)+var2Coefficient*exp(var2*t) ...
+      @ In, container, object, self-like object where all the variables can be stored
+      @ In, Inputs, dict, dictionary of inputs from RAVEN
+
+    """
+    Xi                = np.zeros(container.numberPoints+1)
+    monotonicVariable = np.zeros(container.numberPoints+1)
+    monoVarVal = container.startValue
+    monotonicVariable[0] = container.startValue
+    varCoeff = np.asarray([container.coefficients[var] for var in container.variables])
+    varExponents = np.asarray([Inputs[var]*monoVarVal for var in container.variables])
+    Xi[0] = np.sum(varCoeff*np.exp(varExponents))
+    for step in range(container.numberPoints):
+      monoVarVal+=container.stepSize
+      monotonicVariable[step+1] = monoVarVal
+      varExponents = np.asarray([Inputs[var]*(monoVarVal-monotonicVariable[step]) for var in container.variables])
+      if np.max(varExponents) >= np.finfo(varExponents.dtype).maxexp:
+        print("ExamplePlugin: the exponents of the exponential cause overflow. Increase the number of <numberCalculationPoints>!")
```

Comment:
Unfortunately the Message handler is not passed in.
We might want to automatically pass it... we can discuss this on Monday meeting and open an issue in case?

Reply:
my suggestion would be to pass it automatically but lets discuss it first

Reply:
ok